### PR TITLE
RavenDB-16229 - Memory leak when patching a large collection

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-16229

### CODE REVIEW

File: src/Raven.Server/Documents/CollectionRunner.cs [hunk 1]
```diff
@@ -116,6 +116,8 @@ public virtual Task<IOperationResult> ExecuteDelete(string collectionName, long
                                     break;
                                 }
 
+                                context.Transaction.ForgetAbout(document);
```
Comment (karmeli87) [inline src/Raven.Server/Documents/CollectionRunner.cs:119]: Should we use the `PulsedTransaction` here?

Comment (ayende) [reply src/Raven.Server/Documents/CollectionRunner.cs:119]: That is relevant for compression only, on

Comment (ppekrol) [reply src/Raven.Server/Documents/CollectionRunner.cs:119]: Now I'm wondering. Shouldn't this be after we materialize the Id in ids.Enqueue? We are releasing the buffer before we materialize the Id? Maybe it is not a problem?

Comment (ayende) [reply src/Raven.Server/Documents/CollectionRunner.cs:119]: Yes, need to reverse this, to be safe

Comment (ppekrol) [reply src/Raven.Server/Documents/CollectionRunner.cs:119]: Pushed: https://github.com/ravendb/ravendb/commit/14dc3a84c8d7256374ff3383723e45d367052779

**Pull request merged**